File: internal/config/thinking_test.go
package config

import "testing"

func TestNormalizeThinking(t *testing.T) {
	t.Parallel()
	cases := []struct {
		in, want string
		ok       bool
	}{
		{"", "", true},
		{"  ", "", true},
		{"disabled", "disabled", true},
		{"DISABLED", "disabled", true},
		{"low", "low", true},
		{" medium ", "medium", true},
		{"high", "high", true},
		{"mid", "medium", true},
		{"MID", "medium", true},
		{"enabled", "medium", true},
		{"on", "medium", true},
		{"true", "medium", true},
		{"1", "medium", true},
		{"max", "high", true},
		{"off", "disabled", true},
		{"false", "disabled", true},
		{"0", "disabled", true},
		{"banana", "", false},
		{"medium-high", "", false},
		{"enable", "", false},
	}
	for _, tc := range cases {
		got, ok := NormalizeThinking(tc.in)
		if got != tc.want || ok != tc.ok {
			t.Errorf("NormalizeThinking(%q) = %q, %v; want %q, %v", tc.in, got, ok, tc.want, tc.ok)
		}
	}
}

func TestCanonicalThinking(t *testing.T) {
	t.Parallel()
	if got := CanonicalThinking("enabled"); got != ThinkingMedium {
		t.Errorf("CanonicalThinking(enabled) = %q, want medium", got)
	}
	if got := CanonicalThinking(""); got != "" {
		t.Errorf("CanonicalThinking(\"\") = %q, want empty", got)
	}
	if got := CanonicalThinking("nope"); got != "" {
		t.Errorf("CanonicalThinking(nope) = %q, want empty", got)
	}
	if got := CanonicalThinking("disabled"); got != ThinkingDisabled {
		t.Errorf("CanonicalThinking(disabled) = %q, want disabled", got)
	}
}

File: cmd/odek/ui/js/thinking.test.js
import { test } from 'node:test';
import assert from 'node:assert/strict';
import { normalizeThinking, seedThinking, persistThinking, THINKING_LEVELS } from './thinking.js';

test('normalizeThinking maps aliases and rejects unknown', () => {
  assert.equal(normalizeThinking(''), '');
  assert.equal(normalizeThinking('  LOW '), 'low');
  assert.equal(normalizeThinking('medium'), 'medium');
  assert.equal(normalizeThinking('high'), 'high');
  assert.equal(normalizeThinking('disabled'), 'disabled');
  assert.equal(normalizeThinking('mid'), 'medium');
  assert.equal(normalizeThinking('enabled'), 'medium');
  assert.equal(normalizeThinking('on'), 'medium');
  assert.equal(normalizeThinking('true'), 'medium');
  assert.equal(normalizeThinking('1'), 'medium');
  assert.equal(normalizeThinking('max'), 'high');
  assert.equal(normalizeThinking('off'), 'disabled');
  assert.equal(normalizeThinking('false'), 'disabled');
  assert.equal(normalizeThinking('0'), 'disabled');
  assert.equal(normalizeThinking('banana'), '');
  assert.deepEqual(THINKING_LEVELS, ['disabled', 'low', 'medium', 'high']);
});

test('seedThinking prefers stored, then config string, then disabled', () => {
  assert.equal(seedThinking('low', 'high'), 'low');
  assert.equal(seedThinking('', 'high'), 'high');
  assert.equal(seedThinking('', 'enabled'), 'medium');
  assert.equal(seedThinking('', true), 'medium');
  assert.equal(seedThinking('', false), 'disabled');
  assert.equal(seedThinking('', ''), 'disabled');
  assert.equal(seedThinking('', null), 'disabled');
  assert.equal(seedThinking('nope', 'medium'), 'medium');
});

test('persistThinking writes canonical localStorage', () => {
  const store = new Map();
  globalThis.localStorage = {
    getItem: (k) => (store.has(k) ? store.get(k) : null),
    setItem: (k, v) => store.set(k, String(v)),
    removeItem: (k) => store.delete(k),
  };
  assert.equal(persistThinking('enabled'), 'medium');
  assert.equal(store.get('odek_thinking'), 'medium');
  assert.equal(persistThinking('banana'), 'disabled');
  assert.equal(store.get('odek_thinking'), 'disabled');
});
